ptime.go

```diff
@@ -380,6 +414,23 @@ func (t *Time) Set(year int, month Month, day, hour, min, sec, nsec int, loc *ti
 	if loc == nil {
 		panic("ptime: the Location must not be nil in call to Set")
 	}
+
+	// Normalize nsec, sec, min, hour, overflowing into day.
+	sec, nsec = norm(sec, nsec, 1e9)
+	min, sec = norm(min, sec, 60)
+	hour, min = norm(hour, min, 60)
+	day, hour = norm(day, hour, 24)
+
+	// Normalize month, overflowing into year.
+	m := int(month) - 1
+	year, m = norm(year, m, 12)
+	if divider(25*year+11, 33) < 8 {
```

Comment:
you can use `IsLeap()` method to check for leap years to avoid repetition of the same calculation